### PR TITLE
Check if Stackdriver MetricDescriptor is compatible with the view

### LINKED ISSUE #146: exporter/stackdriver: missing distribution value for CountAggregation
Consider the following view:

```
        view := stats.NewView(
		"my.org/views/video_size_cum",
		"processed video size over time",
		nil,
		videoSize,
		stats.CountAggregation{},
		stats.Cumulative{},
	)
```
It fails with:

2017/11/21 14:52:52 Failed to export to Stackdriver Monitoring: rpc error: code = InvalidArgument desc = Request was missing field timeSeries[0].points[0].distributionValue.

### PR DESCRIPTION
**Pull request by rakyll**
Fixes #146.

### CODE REVIEW

File: exporter/stats/stackdriver/stackdriver.go [hunk 1]
```diff
@@ -309,3 +322,30 @@ func namespacedViewName(v string, escaped bool) string {
...
+	}
+
+	switch md.ValueType {
+	case metricpb.MetricDescriptor_INT64:
+		a = &stats.CountAggregation{}
+	case metricpb.MetricDescriptor_DISTRIBUTION:
+		a = &stats.DistributionAggregation{}
+	}
+
+	if reflect.TypeOf(agg) != reflect.TypeOf(a) {
```
Comment (rakyll) [inline exporter/stats/stackdriver/stackdriver.go]: We should also check whether agg is a pointer or not, or disallow concrete types to be passed as aggregators.

File: exporter/stats/stackdriver/stackdriver.go [hunk 2]
```diff
@@ -309,3 +322,30 @@ func namespacedViewName(v string, escaped bool) string {
...
+		return false
+	}
+	if reflect.TypeOf(window) != reflect.TypeOf(w) {
```
Comment (rakyll) [inline exporter/stats/stackdriver/stackdriver.go]: ditto here.

**Comment by rakyll** [comment]
Added the labels, and addressed the concrete type vs pointers.

File: exporter/stats/stackdriver/stackdriver.go [hunk 3]
```diff
@@ -189,26 +191,36 @@ func (e *Exporter) makeReq(vds []*stats.ViewData) *monitoringpb.CreateTimeSeries
 	}
 }
 
+// createMeasure creates a MetricDescriptor for the given view data in Stackdriver Monitoring.
+// If there is already a metric descriptor created with the same name
+// and the metric descriptor has a different aggregation/window, it returns an error.
```
Comment (songy23) [inline exporter/stats/stackdriver/stackdriver.go]: nit: aggregation/window -> aggregation/window/tag keys

File: exporter/stats/stackdriver/stackdriver.go [hunk 4]
```diff
@@ -189,26 +191,36 @@ func (e *Exporter) makeReq(vds []*stats.ViewData) *monitoringpb.CreateTimeSeries
...
 	m := vd.View.Measure()
 	agg := vd.View.Aggregation()
 	window := vd.View.Window()
+	tagKeys := vd.View.TagKeys()
 	viewName := vd.View.Name()
 
-	_, ok := e.createdViews[viewName]
-	if ok {
+	if md, ok := e.createdViews[viewName]; ok {
+		// Check agg and window.
```
Comment (songy23) [inline exporter/stats/stackdriver/stackdriver.go]: nit: and tag keys

Comment (rakyll) [reply exporter/stats/stackdriver/stackdriver.go]: Done

File: exporter/stats/stackdriver/stackdriver.go (hunk 3 above)
Comment (rakyll) [reply exporter/stats/stackdriver/stackdriver.go]: Done

File: exporter/stats/stackdriver/stackdriver.go [hunk 5]
```diff
@@ -189,26 +191,36 @@ func (e *Exporter) makeReq(vds []*stats.ViewData) *monitoringpb.CreateTimeSeries
...
 	}
 
-	merticName := monitoring.MetricMetricDescriptorPath(e.o.ProjectID, namespacedViewName(viewName, true))
-	_, err := e.c.GetMetricDescriptor(ctx, &monitoringpb.GetMetricDescriptorRequest{
-		Name: merticName,
+	metricName := monitoring.MetricMetricDescriptorPath(e.o.ProjectID, namespacedViewName(viewName, true))
+	md, err := e.c.GetMetricDescriptor(ctx, &monitoringpb.GetMetricDescriptorRequest{
+		Name: metricName,
 	})
+	if err := equalAggWindowTagKeys(md, agg, window, tagKeys); err != nil {
```
Comment (jcd2) [inline exporter/stats/stackdriver/stackdriver.go]: This call to equalAggWindowTagKeys should be moved inside the "if err == nil" block below.

Comment (odeke-em) [reply exporter/stats/stackdriver/stackdriver.go]: I'll complete this part of the review @jcd2, if it is okay with you @rakyll.

Comment (odeke-em) [reply exporter/stats/stackdriver/stackdriver.go]: This was fixed by @songy23 in PR https://github.com/census-instrumentation/opencensus-go/pull/205 so it is all done.

**Pull request merged**